### PR TITLE
Use new detekt features and remove 1 year old versions support

### PR DESCRIPTION
**Pull request by tasomaniac**
This PR removes support for really really old versions. After those versions, the Detekt Gradle plugin was re-written in a much nicer way.

The newest version also supports `failFast` property which is much nicer than to define in `yaml` file. That's why we can override it when available to make sure that our `collectViolations` run.

**Note:** Also fixed bunch of documentation issues.

### CODE REVIEW

File: plugin/src/test/groovy/com/novoda/staticanalysis/internal/detekt/DetektIntegrationTest.groovy [hunk 1]
```diff
@@ -14,15 +15,23 @@ import static com.novoda.test.LogsSubject.assertThat
 class DetektIntegrationTest {
```
Comment (tasomaniac) [inline plugin/src/test/groovy/com/novoda/staticanalysis/internal/detekt/DetektIntegrationTest.groovy]: These look like changes in this test but actually it is just that `DetektIntegrationTest` was removed and `DetektNewIntegrationTest` was renamed.

File: plugin/src/test/groovy/com/novoda/test/TestAndroidKotlinProject.groovy [hunk 2]
```diff
@@ -9,8 +9,8 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.1.4'
-        classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.2.20'
+        classpath 'com.android.tools.build:gradle:3.4.0'
+        classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.31'
```
Comment (tasomaniac) [inline plugin/src/test/groovy/com/novoda/test/TestAndroidKotlinProject.groovy]: I was using Gradle 5 to use locally and that's why updated the Kotlin version.

File: plugin/src/test/groovy/com/novoda/test/TestAndroidKotlinProject.groovy [hunk 3]
```diff
@@ -9,8 +9,8 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.1.4'
-        classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.2.20'
+        classpath 'com.android.tools.build:gradle:3.4.0'
```
Comment (tasomaniac) [inline plugin/src/test/groovy/com/novoda/test/TestAndroidKotlinProject.groovy]: Newer tools have lots of speed improvements.

**Comment by mr-archano** [comment]
sorry @tasomaniac, now that I merged the other PRs this needs some conflict resolution. Could you please take a look at this when you have 5 mins to spare?

**Comment by tasomaniac** [comment]
That's totally fine. I will have a look and actually make sure that detekt tasks are lazily configured. 🖖

File: plugin/src/test/groovy/com/novoda/staticanalysis/internal/detekt/DetektIntegrationTest.groovy [hunk 4]
```diff
@@ -14,15 +14,24 @@ import static com.novoda.test.LogsSubject.assertThat
...
-                [TestProjectRule.forAndroidKotlinProject(), "1.0.0.RC8"],
+                [TestProjectRule.forKotlinProject(), "1.0.0.RC9.2"],
+                [TestProjectRule.forAndroidKotlinProject(), "1.0.0.RC9.2"],
+                [TestProjectRule.forKotlinProject(), "1.0.0-RC10"],
+                [TestProjectRule.forAndroidKotlinProject(), "1.0.0-RC10"],
+                [TestProjectRule.forKotlinProject(), "1.0.0-RC11"],
+                [TestProjectRule.forAndroidKotlinProject(), "1.0.0-RC11"],
+                [TestProjectRule.forKotlinProject(), "1.0.0-RC12"],
+                [TestProjectRule.forAndroidKotlinProject(), "1.0.0-RC12"],
+                // Work fine but requires Gradle 5.x to run
```
Comment (tasomaniac) [inline plugin/src/test/groovy/com/novoda/staticanalysis/internal/detekt/DetektIntegrationTest.groovy]: Added these. Work fine locally for me when I use local installation of Gradle 5.4. 
Unfortunately had to comment our for our 4.10 wrapper.

**Pull request merged**